orchagent/interface_rates.lua

```diff
@@ -0,0 +1,95 @@
+-- KEYS - rif IDs
+-- ARGV[1] - poll time interval
+-- return nothing 
+
+local asic_db = 1 
+local counters_db = 2
+local config_db = 4 
+-- local fc_db = 5
+local counters_table_name = "COUNTERS" 
+-- local asic_table_name = "ASIC_STATE:SAI_OBJECT_TYPE_PORT" 
+local rates_table_name = "RATES"
+-- local fc_group_table_name = "FLEX_COUNTER_GROUP_TABLE:PORT_STAT_COUNTER"
+
+local rets = {}
+
+-- Get configuration
+redis.call('SELECT', config_db)
+local smooth_interval = redis.call('GET', rates_table_name .. ':' .. 'RIF_SMOOTH_INTERVAL')
+local alpha = redis.call('GET', rates_table_name .. ':' .. 'RIF_ALPHA')
+local one_minus_alpha = 1 - alpha
+local delta = ARGV[1]
+
+-- Get speeds
+local speeds = {}
+redis.call('SELECT', asic_db)
+local n = table.getn(KEYS)
+for i = 1, n do
+    speeds[i] = 40000 
+    --redis.call('HGET', counters_table_name .. ':' .. KEYS[i], 'SAI_PORT_ATTR_SPEED')
```

Comment:
commented line is better than speeds[i] = 40000, however it is working only for port RIFs.
For VLAN I believe it should be sum of vlan members, for PortChannel sum of speeds of port channel members.
Same as we do when we display speed for those interfaces in 'show interface status'